lib/improver/utilities/cube_units.py

```diff
@@ -93,28 +94,37 @@ def enforce_units_and_dtypes(cubes, coords=None, enforce=True):
             conforms = _check_units_and_dtype(item, units, dtype)
             if not conforms:
                 msg = ('{} with units {} and datatype {} does not conform'
-                       ' to expected standard (units {}, datatype {})')
+                       ' to expected standard (units {}, datatype {})\n')
                 msg = msg.format(item.name(), item.units, item.dtype,
                                  units, dtype)
-                raise ValueError(msg)
+                error_string += msg
+                continue
 
-        # convert units
+        # attempt to convert units and record any errors
```

Comment:
I still think we need an ```else``` here, otherwise we are back to converting all the coordinates/data even if enforce is False (unless I've missed some cunning somewhere?).

Reply:
The ```continue``` after the error in the ```not``` loop moves us on to the next object in the list.  Although actually, I think I should pull it out to the next indentation level - currently the code will convert units and datatypes to what they are already in, if that makes sense, but there's no need to make those ineffectual calls.

Reply:
Ah yes, that cunningly hidden giant continue statement. Whoops. Thanks.

Reply:
But, yes moving it out a level makes sense.